### ISSUE TITLE
Add a lit-html todomvc demo

### ISSUE THREAD

**Issue by bgrins** [WebKit/Speedometer#52]
As discussed in #3. Unfortunately there's not an example at https://github.com/tastejs/todomvc/tree/master/examples so we may have to write one from scratch or look around elsewhere

**Comment by rniwa** [comment]
@justinfagnani : Any appetite from lit-html community to add this as a benchmark content for Speedometer v3?

Everyone: should this be v3 blocker or is it deferrable to v4?

**Comment by justinfagnani** [comment]
Definitely, lots of appetite!

I'm about to take some vacation, but let me see if the team can jump on this. What's the timeframe for v3 look like?

**Comment by bgrins** [comment]
> There's a PR for Lit here: https://github.com/tastejs/todomvc/pull/2178 but it also includes Redux. I can make a version that doesn't have Redux.

Yes, something like that without Redux sounds nice. Quickly scanning that I believe we've been using a shared CSS file instead of CSS-in-JS but I might be misremembering (cc @flashdesignory who's been driving the version updates and cleanup with the workloads - for example in https://docs.google.com/document/d/1MXxk9DryJmvAmPMt2uMneaS3J5Y-dMFGt4EGvMzhhJE/edit)

> What's the timeframe for v3 look like?

We are aiming for a code freeze on workloads before the end of half.

**Comment by rniwa** [comment]
>
> I'm about to take some vacation, but let me see if the team can jump on this. What's the timeframe for v3 look like?

We're planning to finalize the benchmark content by the end of June.

**Comment by justinfagnani** [comment]
@bgrins 

> Quickly scanning that I believe we've been using a shared CSS file instead of CSS-in-JS

Any library using shadow DOM will need to consume CSS in a way that let's them add it to the shadow roots, and not add it globally. CSS-in-JS is how Lit does this. We also support CSS modules (with import attributes), but that's only available in Chromium browsers right now.

**Comment by bgrins** [comment]
> How that's done, should be framework specific in my opinion.

Sounds good, thanks @justinfagnani!

**Comment by justinfagnani** [comment]
For Speedometer purposes I would much rather do things the idiomatic Lit way - which is to have inline CSS-in-JS scoped to that one shadow root, than try to import a global stylesheet into every shadow root. That will test the characteristics of real-world Lit projects.

I'm not sure if it's possible to automatically pull apart the published CSS into optimized per-component styles in a way that could survive updates. I'll make a PR where I divide up the styles by hand, and we can discuss whether that's ok for the todomvc repo. If not we could try to come up with a way to automatically modularize the CSS, or make a new repo in the Lit org and host it there.

**Comment by julienw** [comment]
I was gonna write that I could live with a lit+Redux implementation if this means we can have this in speedometer v3 as I didn't expect Redux to radically change the performance characteristics of the app. But then I've seen at least one perf problem in the current lit+redux-based PR (I commented there directly). So now I think I'd rather have a version without Redux.

**Comment by flashdesignory** [comment]
@justinfagnani - regarding css-in-js / css modules: 
Here is how it could look like: 
https://github.com/flashdesignory/todomvc-app-css-alt

This was a result of an a11y audit I did on the current todomvc apps for speedometer and a great opportunity to take a look how css modules could look like. 

changes compared to the current state:
https://github.com/flashdesignory/todomvc-app-css-alt/blob/main/CHANGES.md

Let me know if this is helpful. There are some structural changes proposed compared to the official todomvc template, but maybe this version might be a good candidate to try.

**Comment by justinfagnani** [comment]
I'm on vacation for a bit. @rictic is taking over the Lit TodoMVC work. Hopefully we can get this done in time!

**Comment by rictic** [comment]
I've got one up at https://lit.github.io/todomvc/ - one html file, one js module, targeting ES2022.

I didn't notice your comment @flashdesignory until just now so I used the styles that the existing examples at todomvc.com just for convenience, as exactly matching their behavior and look and feel gave a clear success criteria. I definitely agree that the styles are not accessible, I'll take a look at incorporating your improved styles.
